frontend/src/api/doctor.ts

```diff
@@ -0,0 +1,53 @@
+import type { Doctor } from "../components/doctor/DoctorControl";
+const commonRequestOptions = {
+  headers: {
+    Accept: "application/json",
+    "Content-Type": "application/json",
+  },
+};
+
+export const getAllDoctor = async () => {
+  const request = await fetch("/api/admin");
+  const result = await request.json();
+
+  //console.log(result);
+  return result;
+};
+
+export const postDoctor = async (addDoctor: Doctor) => {
+  const requestOptions = {
+    ...commonRequestOptions,
+    method: "POST",
+    body: JSON.stringify(addDoctor),
+  };
+  try {
+    const fetchResponse = await fetch("/api/doctor", requestOptions);
+    console.log(fetchResponse);
+    return await fetchResponse.json();
+  } catch (error) {
+    console.log(error);
+    return error;
```

Comment:
I would `throw new Error(error)` here instead of returning because then where you call this function, you can use `try/catch` blocks to handle errors.

Reply:
`export const postDoctor = async (addDoctor: Doctor) => {
  const requestOptions = {
    ...commonRequestOptions,
    method: "POST",
    body: JSON.stringify(addDoctor),
  };
  try {
    const fetchResponse = await fetch("/api/doctor", requestOptions);
    console.log(fetchResponse);
    await fetchResponse.json();
    if (!addDoctor.firstName) {
      throw new SyntaxError("Incomplete data: no  first name");
    } else if (!addDoctor.lastName) {
      throw new SyntaxError("Incomplete data: no  last name");
    } else if (!addDoctor.email) {
      throw new SyntaxError("Incomplete data: no  email");
    } else if (!addDoctor.specialty) {
      throw new SyntaxError("Incomplete data: no  speciality");
    } else if (!addDoctor.location) {
      throw new SyntaxError("Incomplete data: no  location");
    } else if (!addDoctor.gender) {
      throw new SyntaxError("Incomplete data: no  gender");
    }
  } catch (error) {
    console.log(error);
    return error;
  }
};`

Reply:
Do you mean like this

Reply:
Is that correct?
`export const postDoctor = async (addDoctor: Doctor) => {
  if (!addDoctor.firstName) {
    throw new Error("Incomplete data: no  first name");
  } else if (!addDoctor.lastName) {
    throw new Error("Incomplete data: no  last name");
  } else if (!addDoctor.email) {
    throw new Error("Incomplete data: no  email");
  } else if (!addDoctor.specialty) {
    throw new Error("Incomplete data: no  speciality");
  } else if (!addDoctor.location) {
    throw new Error("Incomplete data: no  location");
  } else if (!addDoctor.gender) {
    throw new Error("Incomplete data: no  gender");
  }
  const requestOptions = {
    ...commonRequestOptions,
    method: "POST",
    body: JSON.stringify(addDoctor),
  };

  try {
    const fetchResponse = await fetch("/api/doctor", requestOptions);
    console.log(fetchResponse);
    await fetchResponse.json();
  } catch (error) {
    console.log(error);
    throw error;
  }
};`

Reply:
I think all of that validation logic should go in the backend. Then in the frontend all you need to do is throw the error.